app/features/build_runner/build_runner.rb

```diff
@@ -173,18 +173,26 @@ def pre_run_action(&completion_block)
 
       checkout_sha do |checkout_success|
         if checkout_success
-          setup_tooling_environment
-          setup_build_specific_environment_variables
+          if setup_tooling_environment # see comment for `#setup_tooling_environment` method
```

Comment:
what happens if this return false? What kind of recovery do we need/error messaging?